msTeams/app.py

```diff
@@ -40,97 +40,116 @@
+
+                    join_btn = meeting_control_panel.find_element_by_xpath("//button[@class='join-btn']")
+                    camera_toggle = WebDriverWait(presentation_container, 10).until(EC.element_to_be_clickable((By.XPATH, "//toggle-button[@data-tid='toggle-video']/div/button")))
+                    mic_toggle = WebDriverWait(presentation_container, 10).until(EC.element_to_be_clickable((By.XPATH, "//toggle-button[@data-tid='toggle-mute']/div/button")))
+
+                    if(camera_toggle.get_attribute("aria-pressed") == 'false'):
+                        camera_toggle.click()
+                    if(mic_toggle.get_attribute("aria-pressed") == 'false'):
+                        mic_toggle.click()
                     join_btn.click()
-                else:
-                    continue
-
-                # Configure presentation options from the meeting lobby. i.e.
-                # Switch off camera and microphone and join the meeting.
-                meeting_container = WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.CLASS_NAME, "video-and-name-input")))
-                meeting_control_panel = meeting_container.find_element_by_xpath("//section[@class='controls-container']")
-                presentation_container = meeting_container.find_element_by_xpath("//div[@role='presentation']")
-
-                join_btn = meeting_control_panel.find_element_by_xpath("//button[@class='join-btn']")
-                camera_toggle = WebDriverWait(presentation_container, 10).until(EC.element_to_be_clickable((By.XPATH, "//toggle-button[@data-tid='toggle-video']/div/button")))
-                mic_toggle = WebDriverWait(presentation_container, 10).until(EC.element_to_be_clickable((By.XPATH, "//toggle-button[@data-tid='toggle-mute']/div/button")))
-
-                if(camera_toggle.get_attribute("aria-pressed") == 'false'):
-                    camera_toggle.click()
-                if(mic_toggle.get_attribute("aria-pressed") == 'false'):
-                    mic_toggle.click()
-                join_btn.click()
-                is_meetings_in_channel = True
-                # FIXME: Should we really break here to not search for further meetings?
+                    is_meetings_in_channel = True
+                    is_meeting_joined = True
+                    # FIXME: Should we really break here and not search in further channels?
+                    break
+            # FIXME: Should we really break here and not search in further teams?
+            if is_meetings_in_channel:
                 break
-        if is_meetings_in_channel:
-            break
-    print('\n')
+        print('\n')
+
+# Run the search for the first time
+search_and_join_meetings()
+
+# If no ongoing meetings found, then re-search after 10 mins(?)
+while not is_meeting_joined:
+    styled_warning("No meetings were found. Re-searching in 600 secs")
+    time.sleep(60)
+    styled_warning("Initiating search...")
+    search_and_join_meetings()
```

Comment:
This logic could be more polished if we allow user interception to re-start the search without waiting for the default time to get over?